editor/sidebar/featured-image/index.js

```diff
@@ -52,19 +53,18 @@ class FeaturedImage extends Component {
 			return;
 		}
 		this.setState( { loading: true } );
-		const mediaIdToLoad = this.props.featuredImageId;
-		this.fetchMediaRequest = new wp.api.models.Media( { id: mediaIdToLoad } ).fetch()
+		if ( this.fetchMediaRequest ) {
+			this.fetchMediaRequest.abort();
+		}
+		this.fetchMediaRequest = new wp.api.models.Media( { id: this.props.featuredImageId } ).fetch()
 			.done( ( media ) => {
-				if ( this.props.featuredImageId !== mediaIdToLoad ) {
-					return;
-				}
 				this.setState( {
 					loading: false,
 					media,
 				} );
 			} )
-			.fail( () => {
-				if ( this.props.featuredImageId !== mediaIdToLoad ) {
+			.fail( ( e ) => {
```

Comment:
I dislike acronyms, especially `e`, because it takes me a moment to distinguish intent between the `event` and `error` common use-cases. I'm always happier with the verbose equivalent.